specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/DatabaseOperations.json

```diff
@@ -0,0 +1,332 @@
+            "CancelInProgress",
+            "Cancelled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ManagementOperationState",
+            "modelAsString": true
+          }
+        },
+        "errorCode": {
+          "format": "int32",
+          "description": "The operation error code.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "errorDescription": {
+          "description": "The operation error description.",
+          "type": "string",
+          "readOnly": true
+        },
+        "errorSeverity": {
+          "format": "int32",
+          "description": "The operation error severity.",
+          "type": "integer",
+          "readOnly": true
+        },
+        "isUserError": {
+          "description": "Whether or not the error is a user error.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "estimatedCompletionTime": {
+          "format": "date-time",
+          "description": "The estimated completion time of the operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "description": {
+          "description": "The operation description.",
+          "type": "string",
+          "readOnly": true
+        },
+        "isCancellable": {
+          "description": "Whether the operation can be cancelled.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "operationPhaseDetails": {
+          "$ref": "#/definitions/PhaseDetails",
+          "description": "The operation phase details.",
+          "readOnly": true
+        }
+      }
+    },
+    "PhaseDetails": {
+      "description": "The phase details properties of a database operation.",
+      "type": "object",
+      "properties": {
+        "phase": {
+          "description": "The operation phase.",
+          "enum": [
+            "Copying",
+            "Catchup",
+            "WaitingForCutover",
+            "CutoverInProgress"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "Phase",
+            "modelAsString": true
+          }
+        },
+        "phaseInformation": {
+          "description": "The operation phase information.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "readOnly": true
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DatabaseNameParameter": {
+      "name": "databaseName",
+      "in": "path",
+      "description": "The name of the database.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ServerNameParameter": {
+      "name": "serverName",
+      "in": "path",
+      "description": "The name of the server.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    }
+  },
```

Comment:
these are also defined in DatabaseRecommendedActions.json. Could you have a single file for these shared types and refer from that ?


This is also the same pattern for other defintions too - they are repeated in each swagger file. Its better to have a common place to define them and then refer from there.